src/App.tsx

```diff
@@ -173,22 +173,33 @@ export class App extends React.Component<RouteComponentProps, State> {
       fetch('/card/read')
         .then(result => result.json())
         .then(card => {
-          const { isVoterCardPresent } = this.state
+          const {
+            isVoterCardPresent,
+            isPollWorkerCardPresent,
+            isClerkCardPresent,
+          } = this.state
+
           if (isVoterCardPresent && !card.present) {
             this.resetBallot()
             return
           }
 
-          if (card.shortValue) {
-            const cardData = JSON.parse(card.shortValue) as CardData
-            this.processCardData({
-              cardData: cardData,
-              longValueExists: card.longValueExists,
-            })
-          } else {
-            this.setState({
-              ...defaultCardPresentState,
-            })
+          if (
+            !isVoterCardPresent &&
+            !isPollWorkerCardPresent &&
+            !isClerkCardPresent
+          ) {
+            if (card.shortValue) {
+              const cardData = JSON.parse(card.shortValue) as CardData
+              this.processCardData({
+                cardData: cardData,
+                longValueExists: card.longValueExists,
+              })
+            } else {
+              this.setState({
```

Comment:
I think this logic is wrong, because what happens when you remove a card? It needs to run this `setState` and I don't think it ever gets here. I'll check the actual running code.

Reply:
yep, confirmed in running code, card removal is no longer processed.

Let's back out this commit to make sure we don't introduce other weird logic issues?